demos/paywall_bot/tests.py

```diff
@@ -0,0 +1,301 @@
+"""The module contains the tests for HammettPaywallBot.
+
+In the case of testing the permission mechanism, it is necessary to create
+a new Screen class in each test case. This is because the Screen object is
+a singleton, and when wrapping all its handlers, they share their state between
+the tests, leading to unstable behavior.
```

Comment:
`When testing the permission mechanism, it's necessary for each test to have its own screen classes. Otherwise, all tests will share the same screens since screens are singletons.`